app/controllers/auth/sessions_controller.rb

```diff
@@ -25,9 +25,11 @@ def new
 
   def create
     super do |resource|
-      resource.update_sign_in!(request, new_sign_in: true)
-      remember_me(resource)
-      flash.delete(:notice)
+      # We only need to call this if this hasn't already been
+      # called from one of the two-factor or sign-in token
+      # authentication methods
+
+      on_authentication_success(resource, :password) unless @on_authentication_success_called
```

Comment:
Interestingly, this didn't call `sign_in(user)` but `on_authentication_success` now does. Could that become an issue?

Reply:
So, looked into it a bit more: `sign_in` is already called in `Devise::SessionsController#create`, so this call is superfluous but it doesn't seem like it could cause issues.